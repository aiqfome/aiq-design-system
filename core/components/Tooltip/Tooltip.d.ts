import React, { ReactNode } from 'react';
export interface Props {
    children?: ReactNode;
    body?: any;
    variant?: 'topLeft' | 'topRight' | 'topCenter' | 'bottomLeft' | 'bottomRight' | 'bottomCenter';
    type?: 'default' | 'balloon';
    onMouseOver?: () => void;
}
export declare const Tooltip: React.FC<Props>;
//# sourceMappingURL=Tooltip.d.ts.map