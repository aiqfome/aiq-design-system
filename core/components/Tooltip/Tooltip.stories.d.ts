import React from 'react';
declare const _default: {
    component: React.FC<import("./Tooltip").Props>;
    title: string;
};
export default _default;
export declare const Basic: React.FC;
export declare const Variants: React.FC;
export declare const Types: React.FC;
export declare const Body: React.FC;
//# sourceMappingURL=Tooltip.stories.d.ts.map