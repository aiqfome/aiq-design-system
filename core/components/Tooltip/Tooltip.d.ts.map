{"version":3,"file":"Tooltip.d.ts","sourceRoot":"","sources":["../../../src/components/Tooltip/Tooltip.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,SAAS,EAAE,MAAM,OAAO,CAAA;AASxC,MAAM,WAAW,KAAK;IACpB,QAAQ,CAAC,EAAE,SAAS,CAAA;IACpB,IAAI,CAAC,EAAE,GAAG,CAAA;IACV,OAAO,CAAC,EACJ,SAAS,GACT,UAAU,GACV,WAAW,GACX,YAAY,GACZ,aAAa,GACb,cAAc,CAAA;IAClB,IAAI,CAAC,EAAE,SAAS,GAAG,SAAS,CAAA;IAC5B,WAAW,CAAC,EAAE,MAAM,IAAI,CAAA;CACzB;AAuBD,eAAO,MAAM,OAAO,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CA6DnC,CAAA"}