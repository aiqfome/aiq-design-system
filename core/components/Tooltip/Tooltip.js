"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tooltip = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const rc_tooltip_1 = __importDefault(require("rc-tooltip"));
const rc_dropdown_1 = __importDefault(require("rc-dropdown"));
const styled_components_1 = __importDefault(require("styled-components"));
const Flex_1 = require("../Flex");
const PopoverStyled = styled_components_1.default(Flex_1.Flex) `
  position: relative;
  margin: 0;
  text-align: left;
  background-color: #fff;
  background-clip: padding-box;
  border-radius: 12px;
  outline: none;
  -webkit-box-shadow: 0px 3px 6px #00000029;
  box-shadow: 0px 3px 6px #00000029;
`;
const TooltipStyled = styled_components_1.default(Flex_1.Flex) `
  position: relative;
  margin: 0;
  text-align: left;
  background-clip: padding-box;
  border-radius: 4px;
  outline: none;
`;
exports.Tooltip = (_a) => {
    var { body, children, type = 'default', variant = 'bottomLeft' } = _a, props = __rest(_a, ["body", "children", "type", "variant"]);
    const child = react_1.default.Children.only(react_1.default.createElement("div", { "data-testid": 'tooltip-child' }, children));
    const getOverlayBalloon = () => (react_1.default.createElement(TooltipStyled, Object.assign({ flexDirection: 'column' }, props), body));
    const getOverlay = () => (react_1.default.createElement(PopoverStyled, Object.assign({ flexDirection: 'column', "data-testid": 'tooltip-content' }, props), body));
    const getTooltipVariant = () => {
        if (variant === 'bottomCenter')
            return 'bottom';
        if (variant === 'topCenter')
            return 'top';
        return variant;
    };
    if (type === 'default') {
        return (react_1.default.createElement(rc_dropdown_1.default, Object.assign({ arrow: false, trigger: ['hover'], placement: variant, overlay: getOverlay, overlayClassName: 'popover' }, props), child));
    }
    return (react_1.default.createElement(rc_tooltip_1.default, Object.assign({ arrowContent: true, trigger: ['hover'], overlayClassName: 'tooltip', overlay: getOverlayBalloon, placement: getTooltipVariant() }, props), child));
};
exports.Tooltip.propTypes = {
    body: prop_types_1.default.any,
    children: prop_types_1.default.node.isRequired,
    type: prop_types_1.default.oneOf(['default', 'balloon']),
    variant: prop_types_1.default.oneOf([
        'topRight',
        'topLeft',
        'topCenter',
        'bottomLeft',
        'bottomRight',
        'bottomCenter'
    ])
};
