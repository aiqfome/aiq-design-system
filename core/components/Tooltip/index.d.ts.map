{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/Tooltip/index.tsx"],"names":[],"mappings":"AAAA,cAAc,WAAW,CAAA"}