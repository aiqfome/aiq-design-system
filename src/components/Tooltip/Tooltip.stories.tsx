import React, { ReactElement } from 'react'

import { Tooltip } from './Tooltip'
import { Text } from '../Text'
import { Flex } from '../Flex'
import { Button } from '../Button'

export default {
  component: Tooltip,
  title: 'Tooltip'
}

export const Basic: React.FC = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Tooltip body="I'm hungry">
        <Text cursor='pointer'>Hamburguer</Text>
      </Tooltip>
    </Flex>
  )
}

export const Variants: React.FC = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Tooltip mx={10} variant='top' body="I'm hungry">
        <Text cursor='pointer'>Top</Text>
      </Tooltip>
      <Tooltip mx={10} variant='bottom' body="I'm hungry">
        <Text cursor='pointer'>Bottom</Text>
      </Tooltip>
      <Tooltip mx={10} variant='left' body="I'm hungry">
        <Text cursor='pointer'>Left</Text>
      </Tooltip>
      <Tooltip mx={10} variant='right' body="I'm hungry">
        <Text cursor='pointer'>Right</Text>
      </Tooltip>
      <Tooltip mx={10} variant='right-bottom' body="I'm hungry">
        <Text cursor='pointer'>Right-Bottom</Text>
      </Tooltip>
      <Tooltip mx={10} variant='left-bottom' body="I'm hungry">
        <Text cursor='pointer'>Left-Bottom</Text>
      </Tooltip>
    </Flex>
  )
}

export const TooltipInBottom: React.FC = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Tooltip mx={10} variant='top' body="I'm hungry">
        <Button variant='outlined' palette='primary'>
          Top Top Top Top Top Top
        </Button>
      </Tooltip>
      <Tooltip mx={10} variant='left' body="I'm hungry">
        <Button variant='outlined' palette='primary'>
          Left Left Left Left Left Left
        </Button>
      </Tooltip>
      <Tooltip mx={10} variant='right' body="I'm hungry">
        <Button variant='outlined' palette='primary'>
          Right Right Right Right Right Right
        </Button>
      </Tooltip>
      <Tooltip mx={10} variant='right-bottom' body="I'm hungry">
        <Button variant='outlined' palette='primary'>
          Right- Right- Right- Right- Right- Right-Bottom
        </Button>
      </Tooltip>
      <Tooltip
        mx={10}
        variant='left-bottom'
        body=' Left- Left- Left- Left- Left- Left-Bottom'
      >
        <Button variant='outlined' palette='primary'>
          Left- Left- Left- Left- Left- Left-Bottom
        </Button>
      </Tooltip>
      <Tooltip mx={10} variant='bottom' left={-0} body='Bottom Bottom Bottom'>
        <Button variant='outlined' palette='primary'>
          Bottom Bottom Bottom Bottom Bottom Bottom
        </Button>
      </Tooltip>
    </Flex>
  )
}

export const Types: React.FC = (): ReactElement => {
  return (
    <Flex variant='fullCentralized'>
      <Tooltip mx={10} type='default' body="I'm hungry">
        <Text cursor='pointer'>Default</Text>
      </Tooltip>
      <Tooltip mx={10} type='balloon' variant='bottom' body="I'm hungry">
        <Text cursor='pointer'>Balloon</Text>
      </Tooltip>
    </Flex>
  )
}

export const Body: React.FC = (): ReactElement => {
  const Menu = () => {
    return (
      <Flex flexDirection='column'>
        <Text>1 Item</Text>
        <Text>2 Item</Text>
        <Text>3 Item</Text>
        <Text>4 Item</Text>
      </Flex>
    )
  }

  return (
    <Flex variant='fullCentralized'>
      <Tooltip mx={10} variant='right-bottom' body={<Menu />}>
        <Text cursor='pointer'>Body</Text>
      </Tooltip>
    </Flex>
  )
}
