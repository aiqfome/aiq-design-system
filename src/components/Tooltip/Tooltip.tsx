import React, { ReactNode, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'

import { position } from 'styled-system'

import styled, { css } from 'styled-components'
import { Box, Props as BoxProps } from '../Box'

export interface Props extends BoxProps {
  children?: ReactNode
  body?: any
  variant?: 'top' | 'right' | 'bottom' | 'left' | 'right-bottom' | 'left-bottom'
  type?: 'default' | 'balloon'
  onMouseOver?: () => void
}

const tooltipVariations = {
  top: css`
    bottom: 100%;
    left: 50%;
  `,
  right: css`
    top: -5px;
    left: 105%;
  `,
  bottom: css`
    top: 100%;
    left: 50%;
  `,
  left: css`
    top: -5px;
    right: 105%;
  `,
  'right-bottom': css`
    top: 100%;
    left: 80%;
  `,
  'left-bottom': css`
    top: 100%;
    right: 80%;
  `
}

const typesVariants = {
  default: css`
    background-color: #fff;
    color: ${({ theme }) => theme.colors.almostBlack};
  `,
  balloon: css<Props>`
    color: #fff;
    background-color: ${({ theme }) => theme.colors.almostBlack};

    &::after {
      content: '';
      position: absolute;
      border-width: 5px;
      border-style: solid;
    }

    ${({ variant, theme }) => {
      switch (variant) {
        case 'top':
          return css`
            bottom: 136%;
            left: 0;
            &::after {
              top: 100%;
              left: 8px;
              border-color: ${theme.colors.almostBlack} transparent transparent
                transparent;
            }
          `
        case 'bottom':
          return css`
            top: 136%;
            left: 0;
            &::after {
              bottom: 100%;
              left: 8px;

              border-color: transparent transparent ${theme.colors.almostBlack}
                transparent;
            }
          `
        default:
          return css`
            top: 136%;
            left: 0;
            &::after {
              bottom: 100%;
              left: 8px;

              border-color: transparent transparent ${theme.colors.almostBlack}
                transparent;
            }
          `
      }
    }}
  `
}

const TooltipStyled = styled(Box)<Props>`
  position: relative;
  display: inline-block;

  &:hover {
    cursor: auto;
  }

  .tooltip {
    visibility: hidden;
    opacity: 0;
    white-space: nowrap;
    text-align: center;
    border-radius: 4px;
    position: absolute;
    z-index: 1;
    box-shadow: 0px 3px 6px #00000029;
    transition: all 0.3s;
    padding: 4px 8px;

    ${({ variant }) => tooltipVariations[variant || 'right']}

    ${({ type }) => typesVariants[type || 'default']}

    ${position}
  }

  &:hover .tooltip {
    visibility: visible;
    opacity: 1;
  }
`

export const Tooltip: React.FC<Props> = ({ body, children, ...props }) => {
  const ref = useRef(document.createElement('div'))

  useEffect(() => {
    fixPositionInWindow()
  }, [])

  function fixPositionInWindow() {
    const { right } = ref.current.getBoundingClientRect()
    if (right > window.innerWidth) {
      ref.current.style.left = `auto`
      ref.current.style.right = `-16px`
    }
  }

  return (
    <TooltipStyled onMouseOver={fixPositionInWindow} {...props}>
      {children}

      <Box refBox={ref} className='tooltip'>
        {body}
      </Box>
    </TooltipStyled>
  )
}

Tooltip.propTypes = {
  body: PropTypes.any,
  children: PropTypes.node,
  type: PropTypes.oneOf(['default', 'balloon']),
  variant: PropTypes.oneOf([
    'top',
    'right',
    'bottom',
    'left',
    'right-bottom',
    'left-bottom'
  ])
}
