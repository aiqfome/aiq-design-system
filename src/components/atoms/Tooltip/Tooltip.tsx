import React, { ReactNode } from 'react'
import PropTypes from 'prop-types'

import styled, { css } from 'styled-components'
import { Box, Props as BoxProps } from '../Box'

export interface Props extends BoxProps {
  children?: ReactNode
  body?: any
  variant?: 'top' | 'right' | 'bottom' | 'left' | 'right-bottom'
}

const tooltipVariations = {
  top: css`
    bottom: 100%;
    left: 50%;
    margin-left: -60px;
  `,
  right: css`
    top: -5px;
    left: 105%;
  `,
  bottom: css`
    top: 100%;
    left: 50%;
    margin-left: -60px;
  `,
  left: css`
    top: -5px;
    right: 105%;
  `,
  'right-bottom': css`
    top: 100%;
    left: 80%;
  `
}

const TooltipStyled = styled(Box)<Props>`
  position: relative;
  display: inline-block;

  &:hover {
    cursor: auto;
  }

  .tooltip {
    visibility: hidden;
    white-space: nowrap;
    background-color: #fff;
    color: ${({ theme }) => theme.colors.almostBlack};
    text-align: center;
    border-radius: 4px;
    position: absolute;
    z-index: 1;
    box-shadow: 0px 3px 6px #00000029;

    ${({ variant }) => tooltipVariations[variant || 'right']}
  }

  &:hover .tooltip {
    visibility: visible;
  }
`

export const Tooltip: React.FC<Props> = ({ body, children, ...props }) => {
  return (
    <TooltipStyled {...props}>
      {children}

      <Box className='tooltip'>{body}</Box>
    </TooltipStyled>
  )
}

Tooltip.propTypes = {
  body: PropTypes.any,
  children: PropTypes.node
}
