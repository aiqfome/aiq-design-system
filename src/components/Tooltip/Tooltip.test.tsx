import React from 'react'

import { render } from '../utils/test/render'

import { Basic } from './Tooltip.stories'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Basic />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
