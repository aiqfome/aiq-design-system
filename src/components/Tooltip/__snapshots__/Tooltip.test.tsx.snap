// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-hKFyIo bnVQwq"
>
  <div
    className="sc-bdnylx gxDRmI sc-gtssRu fcLWwB"
    onMouseOver={[Function]}
  >
    <span
      className="sc-dlnjPT hFJDnJ"
      cursor="pointer"
    >
      Hamburguer
    </span>
    <div
      className="sc-bdnylx gxDRmI tooltip"
    >
      I'm hungry
    </div>
  </div>
</div>
`;
